### PR TITLE
#5 Add the VM api methods

### PR DESCRIPTION
**Pull request by ja6a**

### CODE REVIEW

File: skytap/network.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ type VpnAttachmentNetwork struct {
 	ConfigurationID *string `json:"configuration_id"`
 }
 
-// VPN described a virtual machine attached to an environment.
+// VPN described a virtual machine attached to a network.
```
Comment (pegerto) [inline skytap/network.go]: The comment still not making sense, VPN attached to a network?

File: skytap/vm.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
+package skytap
+
+import "context"
+
+// Default URL paths
+const (
+	vmsLegacyBasePath = "/configurations/"
```
Comment (pegerto) [inline skytap/vm.go:7]: this can be more generic but happy to keep it as its a pattern use in other resources.

File: skytap/vm_test.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
```
Comment (pegerto) [inline skytap/vm_test.go:16]: need to move this response to `testdata/` fixtures 
https://medium.com/@povilasve/go-advanced-tips-tricks-a872503ac859

File: skytap/vm.go (hunk 2 above)
Comment (ja6a) [reply skytap/vm.go:7]: ok

File: skytap/network.go (hunk 1 above)
Comment (ja6a) [reply skytap/network.go]: done

File: skytap/vm_test.go (hunk 3 above)
Comment (ja6a) [reply skytap/vm_test.go:16]: done

**Review by lokeuei — APPROVED** [review]

**Pull request merged**